android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreen.kt

```diff
@@ -0,0 +1,232 @@
+import agency.five.codebase.android.movieapp.ui.component.MovieCard
+import agency.five.codebase.android.movieapp.ui.component.MovieCardViewState
+import agency.five.codebase.android.movieapp.ui.component.MovieCategoryLabel
+import agency.five.codebase.android.movieapp.ui.component.MovieCategoryLabelViewState
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapperImpl
+import agency.five.codebase.android.movieapp.ui.home.mapper.IHomeScreenMapper
+import agency.five.codebase.android.movieapp.ui.theme.MovieAppTheme
+import agency.five.codebase.android.movieapp.ui.theme.spacing
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.sp
+
+private val homeScreenMapper: IHomeScreenMapper = HomeScreenMapperImpl()
+
+val popularCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+    movieCategories = listOf(
+        MovieCategory.POPULAR_STREAMING,
+        MovieCategory.POPULAR_ON_TV,
+        MovieCategory.POPULAR_FOR_RENT,
+        MovieCategory.POPULAR_IN_THEATRES
+    ),
+    selectedMovieCategory = MovieCategory.POPULAR_STREAMING,
+    movies = MoviesMock.getMoviesList()
+)
+
+val nowPlayingCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+    movieCategories = listOf(
+        MovieCategory.NOW_PLAYING_MOVIES,
+        MovieCategory.NOW_PLAYING_TV
+    ),
+    selectedMovieCategory = MovieCategory.NOW_PLAYING_MOVIES,
+    movies = MoviesMock.getMoviesList()
+)
+
+val upcomingCategoryViewState = homeScreenMapper.toHomeMovieCategoryViewState(
+    movieCategories = listOf(
+        MovieCategory.UPCOMING_TODAY,
+        MovieCategory.UPCOMING_THIS_WEEK
+    ),
+    selectedMovieCategory = MovieCategory.UPCOMING_TODAY,
+    movies = MoviesMock.getMoviesList()
+)
+
+@Composable
+fun HomeRoute(
+    onNavigateToMovieDetails: (HomeMovieViewState) -> Unit
+) {
+    var popularViewState by remember { mutableStateOf(popularCategoryViewState) }
+    var nowPlayingViewState by remember { mutableStateOf(nowPlayingCategoryViewState) }
+    var upcomingViewState by remember { mutableStateOf(upcomingCategoryViewState) }
+
+    HomeScreen(
+        popularCategoryViewState = popularViewState,
+        nowPlayingCategoryViewState = nowPlayingViewState,
+        upcomingCategoryViewState = upcomingViewState,
+        onNavigateToMovieDetails = onNavigateToMovieDetails,
+        onCategoryClick = {
+            when (it.itemId) {
+                0, 1, 2, 3 -> popularViewState =
```

Comment:
Use `.ordinal` property of enums to map these integers. 
i.e.
```kotlin
MovieCategory.POPULAR_STREAMING.ordinal
```

Reply:
Took consultations last week and your colleague Matej said that I can use this way or ordinal, whichever I like more. Found it easier like this because I had index issues with .ordinal property.

Reply:
Fine by me. Thanks for the info 👍🏻